tests/material_opaque_test.py

```diff
@@ -0,0 +1,235 @@
+# coding=utf-8
+from honeybee_energy.material.opaque import EnergyMaterial, EnergyMaterialNoMass
+
+import pytest
+
+
+def test_material_init():
+    """Test the initalization of EnergyMaterial objects and basic properties."""
+    concrete = EnergyMaterial('Concrete', 0.2, 0.5, 800, 0.75,
+                              'MediumSmooth', 0.95, 0.75, 0.8)
+    str(concrete)  # test the string representation of the material
+    concrete_dup = concrete.duplicate()
+
+    assert concrete.name == concrete_dup.name == 'Concrete'
+    assert concrete.thickness == concrete_dup.thickness == 0.2
+    assert concrete.conductivity == concrete_dup.conductivity == 0.5
+    assert concrete.density == concrete_dup.density == 800
+    assert concrete.specific_heat == concrete_dup.specific_heat == 0.75
+    assert concrete.roughness == concrete_dup.roughness == 'MediumSmooth'
+    assert concrete.thermal_absorptance == concrete_dup.thermal_absorptance == 0.95
+    assert concrete.solar_absorptance == concrete_dup.solar_absorptance == 0.75
+    assert concrete.visible_absorptance == concrete_dup.visible_absorptance == 0.8
+
+    assert concrete.resistivity == 1 / 0.5
+    assert concrete.u_value == pytest.approx(2.5, rel=1e-2)
+    assert concrete.r_value == pytest.approx(0.4, rel=1e-2)
+    assert concrete.mass_area_density == pytest.approx(160, rel=1e-2)
+    assert concrete.area_heat_capacity == pytest.approx(120, rel=1e-2)
+
+    concrete.r_value = 0.5
+    assert concrete.conductivity != concrete_dup.conductivity
+    assert concrete.r_value == 0.5
+    assert concrete.conductivity == pytest.approx(0.4, rel=1e-2)
+
+
+def test_material_defaults():
+    """Test the EnergyMaterial default properties."""
+    concrete = EnergyMaterial('Concrete [HW]', 0.2, 0.5, 800, 0.75)
+
+    assert concrete.name == 'Concrete HW'
+    assert concrete.roughness == 'MediumRough'
+    assert concrete.thermal_absorptance == 0.9
+    assert concrete.solar_absorptance == concrete.visible_absorptance == 0.7
+
+
+def test_material_invalid():
+    """Test the initalization of EnergyMaterial objects with invalid properties."""
+    concrete = EnergyMaterial('Concrete', 0.2, 0.5, 800, 0.75)
+
+    with pytest.raises(Exception):
+        concrete.name = ['test_name']
+    with pytest.raises(Exception):
+        concrete.thickness = -1
+    with pytest.raises(Exception):
+        concrete.conductivity = -1
+    with pytest.raises(Exception):
+        concrete.density = -1
+    with pytest.raises(Exception):
+        concrete.specific_heat = -1
+    with pytest.raises(Exception):
+        concrete.roughness = 'Medium'
+    with pytest.raises(Exception):
+        concrete.thermal_absorptance = 2
+    with pytest.raises(Exception):
+        concrete.solar_absorptance = 2
+    with pytest.raises(Exception):
```

Comment:
I have been using very generic exceptions in my tests because the main thing I care about is that an exception of some form is raised and I sometimes realize later that the type of error that I raise is not the most appropriate.  I am being kind of lazy here, though, and I should be deliberate about the type of error I raise.  Ok, I will include more specific errors in the next PR.